src/server/interfaces/category.ts

```diff
@@ -0,0 +1,48 @@
+import { Category, CategoryAttributes } from '../models'
+import { logger } from '../services'
+import { ErrorPayload } from '../errorPayload'
+
+export async function getAsync(id: string): Promise<Category | null> {
+    try {
+        const categoryInstance = await Category.getAsync(id)
+
+        return categoryInstance
+    } catch (error) {
+        logger.error(new Error(error))
+        throw error
+    }
+}
+
+export async function findAsync(where: object): Promise<Category[] | null> {
+    try {
+        const categoryInstances = await Category.getManyAsync(where)
+
+        return categoryInstances
+    } catch (error) {
+        logger.error(new Error(error))
+        throw error
+    }
+}
+
+export async function findOneAsync(where: object): Promise<Category | null> {
+    try {
+        const categoryInstances = await Category.getOneAsync(where)
+
+        return categoryInstances
+    } catch (error) {
+        logger.error(new Error(error))
+        throw error
+    }
+}
+
+export async function createAsync(body: CategoryAttributes): Promise<Category | null> {
+    try {
+        const categoryInstance = await Category.createAsync(body)
+        if (!categoryInstance) { throw new ErrorPayload(500, 'Failed to create offer') }
+
+        return getAsync(categoryInstance.id)
```

Comment:
Why are we getting the just created category when we already have it on memory? That was OK when we were making the offers because it has some dependent entities that would be loaded this way. In this case its useless.
Just return the `Category.createAsync(body)`